library/objective-c/EnvoyDeviceLifecycleMonitor.m

```diff
@@ -0,0 +1,35 @@
+#import "library/objective-c/EnvoyEngine.h"
+
+#import "library/common/main_interface.h"
+
+#import <UIKit/UIKit.h>
+
+@implementation EnvoyDeviceLifecycleMonitor
+
++ (void)startFlushingStatsOnLifecycleChanges {
+  static dispatch_once_t lifecycleMonitoringStarted;
+  dispatch_once(&lifecycleMonitoringStarted, ^{
+    [self startObservingLifecycleNotifications];
```

Comment:
I feel sort of like `startObservingLifecycleNotification` should be the public method name, since the fact that we're flushing stats on them seems more like an implementation detail.

Reply:
I can rename